### PR TITLE
Feature/extract last 14 days locations and times & fix daylight saving

### PR DESCRIPTION
**Pull request by AlmaMOH**
Hey,

We're about to launch a feature which extracts a user's last 14 days locations through Google's Timeline.

Here's the branch.

We'd love your feedback / suggestions!

Relates to #92  #76  #112 .

### CODE REVIEW

File: src/services/LocationHistoryService.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        objArray.push(createObject(feature.geometry.coordinates, feature.properties.timespan));
+      } else if (feature.geometry.type === 'LineString') {
+        feature.geometry.coordinates.forEach((point: any) => {
+          objArray.push(createObject(point, feature.properties.timespan));
+        });
+      }
+    }
+  });
+
+  return objArray;
```
Comment (Yshayy) [inline src/services/LocationHistoryService.ts]:
```suggestion
  return  _.flatten(
        features.filter((feature: any) => feature?.properties?.timespan && !feature?.geometry?.coordinates)
        .filter((feature: any) => !config().locationHistoryIgnoreList.includes(feature.properties.Category))
        .map((feature)=>{
            if (feature.geometry.type === 'Point'){
                return [createObject(feature.geometry.coordinates, feature.properties.timespan)]
            }
            if (feature.geometry.type === 'LineString'){
                return feature.geometry.coordinates.map(point => createObject(point, feature.properties.timespan))
            }
            return []
        });
```
Style suggestion, feel free to ignore :)

File: src/components/common/GoogleTimeLine.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    }
+
+    if (data === 'LOGGED_IN' && !isLoggedIn) {
+      try {
+        webViewRef.current.injectJavaScript(`document.getElementsByTagName(\'html\')[0].innerHTML = \'${getLoadingHTML()}\'; true;`);
+
+        setState(prevState => ({ ...prevState, isLoggedIn: true }));
+
+        const kmlUrls = getLastNrDaysKmlUrls();
+
```
[the remark this answers is not in the document]
Comment (greenyossi) [reply src/components/common/GoogleTimeLine.tsx]: Yep, the cookies are saved via the RN webview.

File: src/actions/GeneralActions.ts [hunk 3]
```diff
@@ -43,3 +50,16 @@ const isOlderVersion = (serverVersion: number[], appVersion: number[]) => {
 
   return false;
 };
+
+export const checkIfHideLocationHistory = () => async (dispatch: any) => {
+  try {
+    const firstPointTS = JSON.parse(await AsyncStorage.getItem(FIRST_POINT_TS) || 'false');
+    const shouldHide = await AsyncStorage.getItem(SHOULD_HIDE_LOCATION_HISTORY);
+
+    if (shouldHide || (firstPointTS && (moment().diff(moment(firstPointTS), 'days') > 14))) {
```
[the remark this answers is not in the document]
Comment (greenyossi) [reply src/actions/GeneralActions.ts]: It's taken into consideration and we might add it in future work.. Thanks!

File: src/components/common/GoogleTimeLine.tsx [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        if (pointsData.length === 0) {
+          // once 14 days flow completed for the first time
+          await AsyncStorage.setItem(SHOULD_HIDE_LOCATION_HISTORY, 'true');
+          store().dispatch(checkIfHideLocationHistory());
+          return setState(prevState => ({ ...prevState, openWebview: false, isLoggedIn: false, state: 'successNotFound' }));
+        }
+
+        await insertToSampleDB(pointsData);
+
+        setState(prevState => ({ ...prevState, openWebview: false, isLoggedIn: false, state: 'successFound' }));
```
[the remark this answers is not in the document]
Comment (greenyossi) [reply src/components/common/GoogleTimeLine.tsx]: Great idea! Done :)

File: src/components/common/GoogleTimeLine.tsx [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+        let pointsData: any[] = [];
+
+        if (texts[0].indexOf('DOCTYPE') > -1 && texts[0].indexOf('Error') > -1) {
+          return onFetchError('Fetch KML error');
+        }
+
+        texts.forEach((kml: string) => {
+          pointsData = [...pointsData, ...kmlToGeoJson(kml)];
+        });
```
[the remark this answers is not in the document]
Comment (greenyossi) [reply src/components/common/GoogleTimeLine.tsx]: Done.. Thanks!

File: src/services/LocationHistoryService.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    + '<svg width="94px" height="122px" viewBox="0 0 94 122" version="1.1"> <title>Group 10</title> <desc>Created with Sketch.</desc> <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" opacity="0.298921131"> <g id="No-Signal-Copy" transform="translate(-141.000000, -273.000000)"> <g id="Group-2" transform="translate(91.000000, 237.000000)"> <g id="Group-9" transform="translate(46.000000, 32.000000)"> <g id="Group-3" transform="translate(4.000000, 4.000000)"> <g id="Group-10"> <g id="Group-8"> <g id="Group-7"> <path d="M0,16.0453271 L0,59.8933024 C0,70.1297189 2.63369796,80.2266357 7.62503242,89.2175703 C12.6163669,98.2085048 19.9646208,105.67782 28.6989183,110.934454 L47,122 L47,0 L0,16.0453271 Z" id="Fill-5" fill="#D0D0D0"></path> <path d="M47,0 L47,122 L65.3003646,110.933804 C74.0353792,105.6772 81.3836331,98.0698609 86.3749676,89.2170471 C91.365585,80.3649488 94,70.1285923 94,59.8936666 L94,16.0452331 L47,0 Z" id="Fill-7" fill="#ABABAB"></path> </g> </g> </g> </g> </g> </g> </g> </g> </svg> '
+    + '</div>'
+    + '</body>';
+};
+
+export const getLastNrDaysKmlUrls = () => {
```
[the remark this answers is not in the document]
Comment (greenyossi) [reply src/services/LocationHistoryService.ts]: As mentioned above, currently it's a future plan, thanks!

**Pull request closed without merging**